test/SlowTests/Server/Documents/Revisions/RevisionsReplication.cs

```diff
@@ -927,9 +963,10 @@ public async Task ClearRevisionsFlagAfterExpiration()
 
                 await SetupReplicationAsync(store1, store2);
 
-                WaitForMarker(store1, store2);
+                WaitForMarker(store1, store2, $"marker/{Guid.NewGuid()}$foo/bar");
 
-                var db = await Databases.GetDocumentDatabaseInstanceFor(store1);
+                var dbName = options.DatabaseMode == RavenDatabaseMode.Single ? store1.Database : await Sharding.GetShardDatabaseNameForDocAsync(store1, "foo/bar");
+                var db = await Server.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(dbName);
```

Comment:
can we move this to a method in the infra?

Reply:
done